tsinfer/formats.py

```diff
@@ -2333,6 +2335,19 @@ def sequence_length(self):
     def num_sites(self):
         return self._num_sites
 
+    @functools.cached_property
+    def individuals_mask(self):
+        try:
+            return self.data["samples_mask"][:].astype(bool)
+        except KeyError:
+            return np.full(self._num_unmasked_individuals, True, dtype=bool)
+
+    @functools.cached_property
+    def samples_mask(self):
+        # Samples in sgkit are individuals in tskit, so we need to expand
+        # the mask to cover all the samples for each individual.
+        return np.repeat(self.individuals_mask, self.ploidy)
```

Comment:
Instead of making a new cached array, can't you broadcast a view into the `individuals_mask` instead, so you don't need to make a copy?

Reply:
Hmm, maybe not actually, since this is row-wise. Ignore me.

Reply:
Although I can't actually see anywhere that `samples_mask` is used in the code (although it is in the tests)? m I missing something?

Reply:
Yes, this isn't used directly, but thought it would be useful thing to have.